### PR TITLE
Junius/fix scheduled coldkey swap back unit test

### PR DESCRIPTION
**Pull request by open-junius**
## Related Issue(s)

- Closes #[issue number]

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please describe):

## Breaking Change

If this PR introduces a breaking change, please provide a detailed description of the impact and the migration path for existing applications.

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have run `cargo fmt` and `cargo clippy` to ensure my code is formatted and linted correctly
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## Screenshots (if applicable)

Please include any relevant screenshots or GIFs that demonstrate the changes made.

## Additional Notes

Please provide any additional information or context that may be helpful for reviewers.

### CODE REVIEW

File: pallets/subtensor/src/macros/dispatches.rs [hunk 1]
```diff
@@ -679,13 +677,14 @@ mod dispatches {
 		.saturating_add(T::DbWeight::get().writes(527)), DispatchClass::Operational, Pays::No))]
         pub fn swap_coldkey(
             origin: OriginFor<T>,
+            old_coldkey: T::AccountId,
```
Comment (distributedstatemachine) [inline pallets/subtensor/src/macros/dispatches.rs:680]: Why are you changing the signature of the the swap_coldkey function ? Does the scheduler require it ?

File: pallets/subtensor/src/swap/swap_coldkey.rs [hunk 2]
```diff
@@ -30,12 +30,9 @@ impl<T: Config> Pallet<T> {
     ///
     /// Weight is tracked and updated throughout the function execution.
     pub fn do_swap_coldkey(
-        origin: T::RuntimeOrigin,
+        old_coldkey: &T::AccountId,
```
Comment (distributedstatemachine) [inline pallets/subtensor/src/swap/swap_coldkey.rs:33]: why does this signature have to change?

**Review by distributedstatemachine — COMMENTED** [review]
Please see comments

File: pallets/subtensor/src/macros/dispatches.rs (hunk 1 above)
Comment (open-junius) [reply pallets/subtensor/src/macros/dispatches.rs:680]: I think the extrinsic is just for root. we can't get the old_coldkey from Origin. No Origin can be both root and signed.

Comment (open-junius) [reply pallets/subtensor/src/macros/dispatches.rs:680]: It is not related to scheduler.

File: pallets/subtensor/src/swap/swap_coldkey.rs (hunk 2 above)
Comment (open-junius) [reply pallets/subtensor/src/swap/swap_coldkey.rs:33]: If the function is not a call, no need to use the Origin as argument. And swap_coldkey call it, there is no origin of coldkey. unnessargy to wrap up one for it.

File: pallets/subtensor/src/macros/dispatches.rs (hunk 1 above)
Comment (distributedstatemachine) [reply pallets/subtensor/src/macros/dispatches.rs:680]: makes sense, and the scheduler has root origin ? Just want to be 100% sure only root or the scheduler can call it

Comment (open-junius) [reply pallets/subtensor/src/macros/dispatches.rs:680]: scheduler just accepts the call and trigger it later. scheduler not attach any origin. the root origin is from schedule_swap_coldkey function, subtensor pallet use frame_system::RawOrigin::Root.into() as origin.

T::Scheduler::schedule(
                DispatchTime::At(when),
                None,
                63,
                frame_system::RawOrigin::Root.into(),
                bound_call,
            )

**Pull request merged**